chemie/elections/models.py

```diff
@@ -24,244 +25,316 @@
+                if candidate in ticket.candidates.all():
+                    candidate.votes += 1
+            candidate.save()
+
+    def get_total_votes(self):
+        tickets = self.tickets.all()
+        total_votes = 0
+        candidates = self.candidates.all()
+        for candidate in candidates:
+            total_votes += candidate.pre_votes
+        for ticket in tickets:
+            if ticket.is_blank:
+                total_votes += 1
             else:
-                winners = []
-                all_votes = {}
-                for candidate in self.candidates.all():
-                    all_votes[candidate.id] = candidate.votes
-                winner_spots = self.spots
-                while len(winners) < winner_spots:
-                    most_votes = max(all_votes.values())
-                    winner_ids = []
-                    for candidate_id, votes in all_votes.items():
-                        if votes == most_votes:
-                            winner_ids.append(candidate_id)
-                            # Now set the votes of the last found winner to -1,
-                            # so he is not found again
-                            all_votes[candidate_id] = -1
-                    winner_candidates = Candidate.objects.filter(
-                        id__in=winner_ids
-                    )
-                    winners.extend(list(winner_candidates))
-
-            # All winners are now stored in a list. Add this to self.winners
-            self.winners.add(*winners)
-            """ self.number_of_voters += Profile.objects.filter(voted=True).count() """
-            self.voting_done = True
-            self.save()
-        election = Election.objects.latest("id")
-        election.current_position_is_open = False
-        election.current_position = None
-        election.save()
+                total_votes += ticket.candidates.all().count()
+        return total_votes
+
+    def vote(self, form, user):
+        candidates = form.cleaned_data.get("candidates")
+        ticket = Ticket.create_ticket(candidates) # Lager stemmeseddelen for brukeren
+        self.tickets.add(ticket)
+        user.profile.voted = True
+        user.profile.save()
+        return
+
+    def vote_blank(self, user):
+        blank_ticket = Ticket.create_blank_ticket()
+        self.tickets.add(blank_ticket)
+        user.profile.voted = True
+        user.profile.save()
+        return
 
 
 class Election(models.Model):
     # For the entire election
     is_open = models.BooleanField(verbose_name="Er åpent", default=False)
     positions = models.ManyToManyField(
-        Position, blank=True, related_name="election"
+        Position,
+        blank=True,
+        related_name="election_position",
+        verbose_name="Verv",
     )
     current_position = models.ForeignKey(
         Position,
         blank=True,
         null=True,
         related_name="current_election",
         on_delete=models.CASCADE,
```

Comment:
Om du uheldigvis sletter `current_position` nå, så sletter du uheldigvis hele valget. Kjipt om det skulle skjedd under valg av siste verv. Vurdér om vi skal bruke `models.DO_NOTHING` i stedet for `models.CASCADE`